CDCgov/prime-reportstream
# De-identified schema

This PR adds a schema to de-identify data using RADx MARS requirements:
https://www.nibib.nih.gov/covid-19/radx-tech-program/mars/hl7-message-de-identification-for-sending-to-hhs-protect

Also found other fields potentially containing PHI and removed them as well:
- PID-11.5.1 Patient Address - Zip or Postal Code 
- PID-13.12.1 Phone Number - Home Unformatted Telephone number 
- OBR-22.1.1 Results Rpt/Status Chng - Date/Time
- ORC-23.12.1 Ordering Facility Phone Number - Unformatted Telephone number

Test Steps:
1. Run `./gradlew integrationTests` or run `./prime fhirdata` with a sample file like `./src/testIntegration/resources/datatests/FHIR_to_HL7/sample_CO_20240403.fhir`
2. Review output file and make sure PHI has either been removed or changed to `<deidentified>`.  Use the table provided in https://www.nibib.nih.gov/covid-19/radx-tech-program/mars/hl7-message-de-identification-for-sending-to-hhs-protect to check that it meets the requirements.



## Changes
- added ORU_R01-deidentified schema 
- added integration test for schema

## Checklist

### Testing

### Process
